project/frontend/src/pages/HomePage.js

```diff
@@ -4,23 +4,44 @@ import React, { useState } from 'react';
 
 import Modal from '../components/Modal.js';
 import PreferenceForm from '../components/PreferenceForm.js';
+import LocationFinder from '../components/LocationFinder';
 
 function HomePage(props) {
   const [isPreferenceFormOpen, setIsPreferenceFormOpen] = useState(false);
+  const [isLocationFormOpen, setIsLocationFormOpen] = useState(false);
 
   const togglePreferenceForm = () => {
     setIsPreferenceFormOpen((prev) => !prev);
   };
 
+  const toggleLocationForm = () => {
+    setIsLocationFormOpen((prev) => !prev);
+  };
+
+  const handleClick = () => {
+    togglePreferenceForm();
+    toggleLocationForm();
+  };
+
+  const handleData = (data) => {
+    if (data) {
+      props.history.push(data);
+    }
+  };
+
   return [
     <div key='home-page' className='container u-full-width'>
       <div className='row'>
         <div id='welcome' className='column'>
           <h1 id='logo'>[LOGO]</h1>
           <h4>Discover your restaurant match</h4>
-          <button>Location</button>
+          <button onClick={toggleLocationForm}>Location</button>
           <div>
-            <button onClick={togglePreferenceForm}>Find My Match</button>
+            <button
+              disabled={!Boolean(props.history.location.currLocation)}
```

Comment:
So this could instead just be `disabled={!Boolean(location.currLocation)}`.

Also is the `Boolean()` constructor necessary here? Or is it fine to just do `!location.currLocation` and have the `!` make the expression resolve to a boolean?

Reply:
That makes sense and I think it's needed but can check